ogr/services/github/check_run.py

```diff
@@ -0,0 +1,365 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import datetime
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+
+from github.CheckRun import CheckRun
+from github.GithubObject import NotSet
+
+from ogr.abstract import OgrAbstractClass
+from ogr.exceptions import OperationNotSupported
+from ogr.services import github as ogr_github
+
+GithubCheckRunOutput = Dict[str, Union[str, List[Dict[str, Union[str, int]]]]]
+
+
+class GithubCheckRunStatus(Enum):
+    """
+    Represents statuses GitHub check run can have.
+    """
+
+    queued = "queued"
+    in_progress = "in_progress"
+    completed = "completed"
+
+
+class GithubCheckRunResult(Enum):
+    """
+    Represents conclusion/result of the GitHub check run.
+    """
+
+    action_required = "action_required"
+    cancelled = "cancelled"
+    failure = "failure"
+    neutral = "neutral"
+    success = "success"
+    skipped = "skipped"
+    stale = "stale"
+    timed_out = "timed_out"
+
+
+def value_or_NotSet(value: Optional[Any]) -> Any:
+    """
+    Wrapper for PyGithub, allows us to transform `None` into PyGithub's `NotSet`.
+
+    Args:
+        value: Value that can be None.
+
+    Returns:
+        If value is not None, value is returned; NotSet otherwise.
+    """
+    return value if value is not None else NotSet
+
+
+def create_github_check_run_output(
+    title: str,
+    summary: str,
+    text: Optional[str] = None,
+    annotations: Optional[List[Dict[str, Union[str, int]]]] = None,
+) -> GithubCheckRunOutput:
+    """
+    Helper function for constructing valid GitHub output for check run.
+
+    Args:
+        title: Title of the output.
+        summary: Summary of the output.
+        text: Optional text for the output. Can be markdown formatted.
+
+            Defaults to `None`.
+        annotations: Optional annotations that are tied to source code.
```

Comment:
Is the `Defaults to ..` related to `annotations or `text`?

Reply:
`text` It takes double `\n\n` like GitHub to be put on separate line.

Reply:
Aha, I need to get used to this format.